### ISSUE TITLE
Feature Request: Call for missing documentation

### ISSUE THREAD

**Issue by SimonDarksideJ** [MonoGame/MonoGame#7693]
As the MonoGame team prepares for upcoming releases, it is good to get feedback from the community as to any pains related to documentation and note any gaps / questions developers have had while building projects.

As a starter, here is an initial list based on recent issues raised:

* How to take a MonoGame game built for one platform to another (adding platforms)
* How to setup your project for multi-platform in the beginning
* A simple 2D starter game project (documenting the MonoGame.Samples 2D project)
* A simple 3D starter game project

If the community has other requests, please comment on them below, detailing:

* What the ask is - what do you need to gain from a new doc
* What you would like to see - the kinds of information you think would be helpful
* Additional detail

Not saying we will get to everything but best efforts will be made for the next release to got some more focus on docs

**Comment by Peewi** [comment]
I have noticed that some public classes and methods are missing xml documentation that have it in Microsoft's XNA documentation.

Unless it's a thing that has changed significantly, is there any reason the XNA documentation couldn't be copied?

There are also methods new to Monogame that are undocumented, such as `ContentManager.LoadLocalized`. Although it is described on [a page that is about localization in general](https://docs.monogame.net/articles/content/localization.html).

Basically anytime I have to google something that could have been in intellisense I get a little annoyed.

**Comment by tomspilman** [comment]
> is there any reason the XNA documentation couldn't be copied?

The XNA documentation is copyright Microsoft... we cannot legally copy it.

If someone submits a PR for docs with XNA docs copied over we have to reject it.

**Comment by Peewi** [comment]
I suppose that makes sense in some way, even if I think it's a little weird when the APIs are mostly the same.

**Comment by tomspilman** [comment]
> even if I think it's a little weird when the APIs are mostly the same.

[APIs cannot be copyrighted](https://www.jdsupra.com/legalnews/supreme-court-copying-apis-in-software-8379806/), but documentation can be.

Weird, but true.

**Comment by mrhelmut** [comment]
Dean managed to get Microsoft's approval to use [the XNA documentation](https://docs.microsoft.com/en-us/previous-versions/windows/xna/bb200104(v%3dxnagamestudio.41)), as long as what we grab from it has this copyright to it: ```© 2012 Microsoft Corporation. All rights reserved.```
If need be, we have archives of the API documentation and online articles.

I'll try to come up with a guideline page on how to import the documentation to make sure that we are respecting the copyrights.
We should also likely mention on doc pages with this copyright that Microsoft are the original authors, but that MonoGame isn't Microsoft to avoid any users being mislead into believing that we are Microsoft.

**Comment by mrhelmut** [comment]
The footer indeed looks like the best solution.

**Comment by SimonDarksideJ** [comment]
Well, we have two standard footers with the MS Docs migration

For MS original docs
> © 2012 Microsoft Corporation. All rights reserved.  
> 
> © 2023 The MonoGame Foundation.


But for MonoGame foundation only docs, it is simply

> © 2023 The MonoGame Foundation.

**Comment by mrhelmut** [comment]
> @mrhelmut Was there ever any guideline done for this or is there work being done on this already by @SimonDarksideJ and @dellis1972 in regards to the new documentation stuff they are doing?

Nothing discussed or being worked on in that regard. Suggestions and contributions more than welcome.

**Comment by AristurtleDev** [comment]
> Well, we have two standard footers with the MS Docs migration
> 
> For MS original docs
> 
> > © 2012 Microsoft Corporation. All rights reserved.
> > © 2023 The MonoGame Foundation.
> 
> But for MonoGame foundation only docs, it is simply
> 
> > © 2023 The MonoGame Foundation.

This is gonna get tricky though for some member types like 

https://github.com/MonoGame/MonoGame/blob/225045b3f1579078757a1a63002d764a87e04fa8/MonoGame.Framework/Color.cs#L1764

If we use the MSDN documentation for that Multiply operator member, then we have an instance where documentation is both MonoGame and Microsoft copyright.  This is no longer an MS original doc or a MonoGame original doc.

So for these would you include both in the footer?
Or would it be better to add the copyright as a `<remark>` or other tag within the XML documentation?

**Comment by mrhelmut** [comment]
What about adding both copyright to each and every pages of the documentation?

**Comment by AristurtleDev** [comment]
> What about adding both copyright to each and every pages of the documentation?

Yea, this is what @tomspilman was suggesting here https://github.com/MonoGame/MonoGame/issues/7693#issuecomment-1117520161

**Comment by AristurtleDev** [comment]
Would the original permission to use the Microsoft documentation allow us to do something like the following?

1. Generate the original XML documentation from XNA
2. Put that XML document in the [Depndencies repo](https://github.com/MonoGame/MonoGame.Dependencies)
3. Use the [`<include file='filename' path='tagpath[@name="id"]' />`](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/xmldoc/recommended-tags#include) for members missing XML documentation so that it just pulls it from the XML file without us having to manually copy/paste it over to everything?

Doing it this way would mean we are using the original documentation and making minimal changes in the current repo while also allowing it to be easy to see where the original documentation was used so we know in the future what to update should we want to remove that dependency and have everything 100% MonoGame documentation.

**Comment by mrhelmut** [comment]
That seems relevant to avoid too much work!

I'm not sure of the scope of what they agreed on, but it can't harm to double check. @dellis1972 ?

**Comment by tomspilman** [comment]
If we're talking reference API docs they also help when writing code via intellisense.  So i think we actually need the documentation added to the class/member/etc for that to work.

**Comment by AristurtleDev** [comment]
> If we're talking reference API docs they also help when writing code via intellisense. So i think we actually need the documentation added to the class/member/etc for that to work.

You can use the `<include>` tag like I mentioned and it will pick that up automatically for the intellesense.  Here is an example

**Without the include**
![Without Include](https://github.com/MonoGame/MonoGame/assets/103014489/e111bd90-186d-4ca2-9024-fcc5eed4a72c)

**With Include**
![With Include](https://github.com/MonoGame/MonoGame/assets/103014489/93493f1b-27cb-410f-a985-f368132ab95d)

And this was just by adding the following as the xml documentation for the class

```xml
/// <include file='../ThirdParty/Dependencies/Microsoft.Xna.Framework.xml' path='doc/members/member[@name="T:Microsoft.Xna.Framework.Content.ContentManager"]'/>    
```

It's just pulling the documentation from the xml file

**Comment by AristurtleDev** [comment]
That documentation is pulled and then included in the resulting MonoGame.Framework.xml output 
![image](https://github.com/MonoGame/MonoGame/assets/103014489/ad7ed556-6399-4e3f-bc1a-03f025222a4e)

**Comment by dellis1972** [comment]
> That seems relevant to avoid too much work!
> 
> I'm not sure of the scope of what they agreed on, but it can't harm to double check. @dellis1972 ?

They "permisison" was for the docs they specifically gave us access to was my understanding. I have an archive of that, so I'd need to go through it and see if that included the API docs. I know it included tutorials as Simon (and I) have been porting those over.

**Comment by AristurtleDev** [comment]
> The "permisison" was for the docs they specifically gave us access to was my understanding.

Alright. For what I was suggesting we would need the generated xml from the build of XNA so if that's not included then my idea is a no go unfortunately

**Comment by dellis1972** [comment]
> > The "permisison" was for the docs they specifically gave us access to was my understanding.
> 
> Alright. For what I was suggesting we would need the generated xml from the build of XNA so if that's not included then my idea is a no go unfortunately

So I checked the documentation zip and it does contain API documentation. However it does not contain a doc xml file for the api. The zip is all markdown and toc.yml files, it was basically a code dump from the xna website.

**Comment by AristurtleDev** [comment]
@dellis1972 that's a bummer then. Would have been really simple to update the repo if they had included the XML, but it is what it is.

Would you be able to post/share the API docs they provided? If so I can start getting the work done with other community memebers of copying the API text into the repo for code members that are missing XML documentation.

**Comment by dellis1972** [comment]
Perhaps I can create a temporary private repo with this data in it and give specific people access. I'd rather not post it publicly just to be on the safe side.

**Comment by SimonDarksideJ** [comment]
Agreed, updating the api docs is a completely separate task.
Although given how much of the MonoGame api is already documented, need to ensure we are adding value in updating it

**Comment by AristurtleDev** [comment]
@SimonDarksideJ  there are almost 900 members missing XML documentation. Here is a tracking spreadsheet I setup from a build output

https://docs.google.com/spreadsheets/d/1CMKClYiAuwbw1M7vMwqQFRj-Loq6a7VmTpqbb1HiNTE/edit?usp=sharing

The value added would be so that it's fully documented and we don't have to tell people to just Google the XNA docs like what happened here yesterday https://discord.com/channels/355231098122272778/357384790158213120/1204074211140706334

![image](https://github.com/MonoGame/MonoGame/assets/103014489/d7420e59-4d70-46be-9115-70f57a45e625)

There's no guarantee that I'm aware of that the old XNA documentation site will stay up. If MS ever decided to take it down, we'd then have to revert to using wayback links from internet archive.

**Comment by AristurtleDev** [comment]
I've created Issue #8165 to track the specific discussion of updating the XML documentation

**Comment by SimonDarksideJ** [comment]
All work completed (until the next batch)
